lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,332 @@
+#ifndef __BMSDriverGroup_H__
+#define __BMSDriverGroup_H__
+/**
+ * PREAMBLE:
+ * This file is designed to hold all of the functions that will allow communication with the Teensy.
+ * There is a specific command format that needs to be sent on SPI for this device to respond.
+ * I will do my best to provide background information to all of these commands as I make them.
+ * But for more information, please reference the ACU Firmware Migration on Bookstack
+ * https://wiki.hytechracing.org/books/ht09-design/page/acu-firmware-migration
+ * Link to the LTC6811 data sheet: https://www.analog.com/media/en/technical-documentation/data-sheets/LTC6811-1-6811-2.pdf
+ *
+ * NOTE: Because the broadcast command will provide 8 bytes of data per command call from EACH IC,
+ * Each instance LTC6811 will represent functionality for all ICs on a particular chip select
+ * Therefore, the data stored in one instance will only hold HALF. One for chip_select 9 and the other for 10
+ * During instantiation, we need to make sure that the addresses are assigned for each LTC6811
+ */
+
+/* Library Includes */
+/**
+ * INCLUDE: LTC6811-1.h and LTC6811-2.h for model-specific data packaging
+ * LTC6811-1 can only function through broadcasted commands and therefore used daisy-chained
+ * LTC6811-2 can operate under broadcast and address functionality, but will only perform address-based commands
+ * Irregardless of which model we choose to use, we need will need access to Arduino and SPI repositories.
+ */
+#include <Arduino.h>
+#include <SPI.h>
+#include <string.h>
+#include <stdio.h>
+#include "Configuration.h"
+#include "LTCSPIInterface.h"
+
+class BMSDriverGroup
+{
+public:
+    BMSDriverGroup() = default;
+    BMSDriverGroup(int cs, int ic_num) : chip_select(cs),
+                                         ic_count(ic_num) {};
+
+    /* -------------------- SETUP FUNCTIONS -------------------- */
+
+    /**
+     * INIT: ONLY CALLED ONCE at initialization. LTC6811 only needs to set up the PEC table
+     * @post We should have initialized the PEC by calling generate_PEC_table(), so every other time
+     * we need to find the PEC it will be much faster.
+     */
+    void init();
+
+    /**
+     * PEC:
+     * The Packet Error Code (PEC) is a Error Checker–like CRC for CAN–to make sure that command and data
+     * bytes are communicated to the device correctly.
+     * This code was taken directly from the data sheet, page 76
+     * @post Instantiate the public pec15Table array of size 256. This variable should never be modified afterwards
+     */
+    void generate_PEC_table();
```

Comment:
please make this a private member function. no need for anything else to call this directly